Secret-agency-service/src/test/java/cz/fi/muni/pa165/secretagency/service/facade/AgentFacadeTest.java

```diff
@@ -0,0 +1,139 @@
+package cz.fi.muni.pa165.secretagency.service.facade;
+
+import cz.fi.muni.pa165.secretagency.dto.AgentCreateDTO;
+import cz.fi.muni.pa165.secretagency.dto.AgentDTO;
+import cz.fi.muni.pa165.secretagency.entity.Agent;
+import cz.fi.muni.pa165.secretagency.enums.AgentRankEnum;
+import cz.fi.muni.pa165.secretagency.enums.LanguageEnum;
+import cz.fi.muni.pa165.secretagency.facade.AgentFacade;
+import cz.fi.muni.pa165.secretagency.service.AgentService;
+import cz.fi.muni.pa165.secretagency.service.BeanMappingService;
+import cz.fi.muni.pa165.secretagency.service.config.ServiceConfiguration;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertTrue;
+import static org.testng.AssertJUnit.assertEquals;
+
+/**
+ * Author: Adam Kral <433328>
+ * Date: 11/25/18
+ * Time: 3:57 PM
+ */
+@ContextConfiguration(classes = ServiceConfiguration.class)
+public class AgentFacadeTest extends AbstractTestNGSpringContextTests {
```

Comment:
Já bych testy na téhle vrstvě psal tak, abychom ověřili, že dané operace skutečně fungují. Tudíž už bych nic nemockoval. Myslím, že o tom, jak by měly tyhle testy vypadat ale v zadání nic není.

Reply:
no je tam napsano ze to maji byt Unit testy, coz jsou, pokud by se to nemockovalo, tak jsou to podle me spis integracni testy

Reply:
A jo, máš pravdu. Unit testy na Facade vrstvě jsou ale k ničemu, ne? 😄 Vždyť ty metody nemají kromě mapování žádnou logiku.

Reply:
No je to trochu pro dementy, ale tak v budoucnu te to muze zachranit pred nejakym bugem. Aspon to kontroluje, jakou metodu te servicy volas a ze vracis to same co ti vratila ona